### DISCUSSION TITLE
while loop or recursion or mutable variables

### DISCUSSION THREAD

**Discussion by dailydaniel** [langgenius/dify#6248] [Suggestion]
It will be very useful to have one of the functions:
- ability to change variables (now all of them are immutable)
- ability to use loops while
- ability to use recursion or loop the workflow through a variable aggregator (for now recursion works bad because of max depth 5)

I encountered a problem - it is impossible to use more complex pipelines with iterative refinement of the answer or early completion of the work. The options listed above (any of them) may help with this issue.

**Comment by crazywoola** [comment]
We are working on this feature right now, but still need some time to design this function.

**Reply by mkaskov** [reply]
That's great! Glad to hear it! Which implementation did you choose?

**Reply by suwubee** [reply]
> We are working on this feature right now, but still need some time to design this function.

There are two important links that are currently missing. 
1. Wait for the user to confirm that the previous process is correct before executing the next process. Otherwise, it is meaningless to continue executing. 
2. If the previous process is not satisfactory, re-execute the previous process through judgment, and even add additional instructions.

**Comment by wuheidechuan** [comment]
I also need this functionality. Sometimes the output of large models is not satisfactory. After evaluation, if it doesn't meet the requirements, the large model needs to regenerate the output. This scenario is very useful. I'm glad to hear that this feature is being implemented.
Great！
